helix-core/src/main/java/org/apache/helix/task/TaskStateModelFactory.java

```diff
@@ -176,4 +167,21 @@ protected static RealmAwareZkClient createZkClient(HelixManager manager) {
         new HelixZkClient.ZkConnectionConfig(zkAddress),
         clientConfig.createHelixZkClientConfig().setZkSerializer(new ZNRecordSerializer()));
   }
+
+  private static ScheduledExecutorService createThreadPoolExecutor(HelixManager manager) {
+    RealmAwareZkClient zkClient = createZkClient(manager);
+    ScheduledExecutorService executorService = Executors.newScheduledThreadPool(TaskUtil
+            .getTargetThreadPoolSize(zkClient, manager.getClusterName(), manager.getInstanceName()),
+        new ThreadFactory() {
+          private AtomicInteger threadId = new AtomicInteger(0);
+
+          @Override
+          public Thread newThread(Runnable r) {
+            return new Thread(r, "TaskStateModelFactory-task_thread-" + threadId.getAndIncrement());
+          }
+        });
+    zkClient.close();
```

Comment:
Use try..finally to make sure the client is close, even any exception is thrown before the close.